### PR TITLE
Handle USER in the RUN command

### PR DESCRIPTION
**Pull request by Rowern**
With this PR, after a USER step is done, any subsequent RUN command will use the "saved" USER.

### CODE REVIEW

**Comment by yiranwang52** [comment]
Instead of adding one more variable, `USER` was already persisted in previous step's image config. Just need to make use of it in `ApplyCtxAndConfig`

**Comment by Rowern** [comment]
Thanks for the feedback! I did that, I'm waiting for this to be reviewed and merged to flesh more work on the first PR (handling chown). I saw some other case where USER is not presisted like docker does (a COPY step after a USER change should set the right of the copied file to the user).

**Comment by Rowern** [comment]
I see that tests are failing, I'm looking at why 😄

**Comment by Rowern** [comment]
I forgot that by overriding `ApplyCtxAndConfig` the code setting up the workingdir would fail.

@yiranwang52 I'm not sure of the impact of not having the "Set env from previous stage" not being called before the run step. I think it would not be too impactful but I'm not sure...

**Comment by Rowern** [comment]
I do have my answer with a simple Dockerfile test:

```Dockerfile
FROM alpine

ARG HELLO=1

RUN env
```

If we do not run the set env code, `env` does not contain `HELLO` (but that's logic) so I'm also exporting the code to a method in base_step

File: lib/shell/cmd.go [hunk 1]
```diff
@@ -17,20 +17,48 @@ package shell
...
+
+	if user != "" {
+		// Set the user to the one specified before
+		uid, gid, err := utils.ResolveChown(user)
+		if err != nil {
+			return fmt.Errorf("cmd user resolve: %s", err)
+		}
+
+		uid32 := uint32(uid)
+		gid32 := uint32(gid)
```
Comment (apourchet) [inline lib/shell/cmd.go]: Don't really need those variables here.

File: lib/shell/cmd.go [hunk 2]
```diff
@@ -17,20 +17,48 @@ package shell
...
 	cmd := exec.Command(cmdName, cmdArgs...)
 	if workingDir != "" {
 		cmd.Dir = workingDir
 	}
 
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+
+	currentEnv := os.Environ()
+
+	if user != "" {
```
Comment (apourchet) [inline lib/shell/cmd.go]: This functions getting real long, can you split this out into a function of its own? It also does 2 things:
1) Finds the syscall.Credential
2) Creates the HOME env variable
We could split those out as well.

**Pull request merged**